### PR TITLE
Add invoice/estimate/payment number length setting

### PR DESCRIPTION
**Pull request by TheZoker**
Adds a new option to customize the amount of leading zeros for new invoices/estimates/payments.
Setting this option to `4` will lead to numbers like `INV-0001`. Setting this option to `6` (the default), will lead to numbers `INV-000001`.

Screenshot_2021 03 22_13h58m58s_017_
Screenshot_2021 03 22_13h59m12s_018_

Closes https://github.com/bytefury/crater/issues/382

### CODE REVIEW

File: database/migrations/2021_03_23_145012_add_number_length_setting.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        $user = User::where('role', 'super admin')->first();
+
+        if ($user) {
+            $invoice_number_length = CompanySetting::getSetting('invoice_number_length', $user->company_id);
+            if(empty($invoice_number_length)) {
+                CompanySetting::setSetting('invoice_number_length', '6', $user->company_id);
+            }
+
+            $estimate_number_length = Setting::getSetting('estimate_number_length', $user->company_id);
```
Comment (mohitpanjwani) [inline database/migrations/2021_03_23_145012_add_number_length_setting.php:24]: @TheZoker Hey, I think you missed using CompanySetting on estimate_number_length.

File: resources/assets/js/views/settings/customization-tabs/EstimatesTab.vue [hunk 2]
```diff
@@ -14,6 +14,19 @@
         />
       </sw-input-group>
 
+      <sw-input-group
+        :label="$t('settings.customization.estimates.estimate_number_length')"
+        :error="estimateLeadingZerosError"
```
Comment (mohitpanjwani) [inline resources/assets/js/views/settings/customization-tabs/EstimatesTab.vue:19]: can you change the label to something like `estimateNumberLengthError`

File: resources/assets/js/views/settings/customization-tabs/InvoicesTab.vue [hunk 3]
```diff
@@ -14,6 +14,19 @@
         />
       </sw-input-group>
 
+      <sw-input-group
+        :label="$t('settings.customization.invoices.invoice_number_length')"
+        :error="invoiceLeadingZerosError"
```
Comment (mohitpanjwani) [inline resources/assets/js/views/settings/customization-tabs/InvoicesTab.vue:19]: Same thing here.

File: resources/assets/js/views/settings/customization-tabs/PaymentsTab.vue [hunk 4]
```diff
@@ -15,6 +15,19 @@
         />
       </sw-input-group>
 
+      <sw-input-group
+        :label="$t('settings.customization.payments.payment_number_length')"
+        :error="paymentLeadingZerosError"
```
Comment (mohitpanjwani) [inline resources/assets/js/views/settings/customization-tabs/PaymentsTab.vue:20]: and here

**Review by mohitpanjwani — CHANGES_REQUESTED** [review]
Hey, Thanks for the updates. 

Found a few minor things. 

Also can you regenerate the public files after merging with latest `master` to fix the conflicts?

**Comment by TheZoker** [comment]
Thanks for the review, addressed all comment, rebased onto master and pushed new build files :)

**Pull request merged**